packages/protocol/thread/pause.ts

```diff
@@ -211,6 +207,15 @@ export class Pause {
   addData(...datas: PauseData[]) {
     datas.forEach(d => this._addDataObjects(d));
     datas.forEach(d => this._updateDataFronts(d));
+
+    const pauseId = this.pauseId!;
+    const point = this.point!;
+
+    if (pauseDataHandlers.length > 0) {
+      datas.forEach(data => {
+        pauseDataHandlers.forEach(handler => handler(pauseId, point, data));
+      });
+    }
```

Comment:
This (`addData`) is a better place to call pre-cache because more places may call this than just the `create` and `instantiate` methods.